### PR TITLE
Tutorial k8s

### PR DESCRIPTION
**Pull request by ClaraLavina**
## Done

* created a new k8s tutorial

## QA

* run locally
* read through

### CODE REVIEW

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+### What you’ll need
+
+
+* You have already deployed your [production cluster Kubernetes](https://jujucharms.com/canonical-kubernetes/) with JAAS into your preferred cloud.
+* You have provided your SSH public key to control your cluster. [If you didn’t provide SSH public key earlier](https://jujucharms.com/docs/2.1/users-auth#credentials-and-ssh-keys).
+* Signing in with an Ubuntu One account is required.
+* The Juju command line client for Ubuntu, Windows, macOS or CentOS. You can find more info [here](https://jujucharms.com/docs/stable/getting-started#use-the-command-line).
+* The Kubectl command line. Learn more [here](https://kubernetes.io/docs/user-guide/kubectl-overview/).
+* Some very basic knowledge of command line use, and how to edit files.
```
Comment (anthonydillon) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: A great list of future tutorials.

**Comment by anthonydillon** [comment]
Approved by awaiting a second :+1: from the JAAS team.

**Comment by anthonydillon** [comment]
As this tutorial has had a number of updates since it's review. It will need to buy reviewed again. Please comment when this PR is ready for review.

**Comment by WillMoggridge** [comment]
@anthonydillon Some changes have been pushed up, did you want to review them before we merge?

**Comment by anthonydillon** [comment]
@WillMoggridge Sure I can review but I believe this should get a second +1 from a member of the JAAS team.

**Comment by ClaraLavina** [comment]
@mitechie could you please have a quick look before merging? Thanks!

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+ 
+# Get started with Canonical Kubernetes
+ 
+ 
+## Overview
+Duration: 1:00
+ 
+### About the Canonical Distribution of Kubernetes
+ 
+This solution is a highly available (HA) production ready Kubernetes cluster — with 2 masters, 3 workers, 3 etcd nodes and a load balancer for the HA control plane. It includes logging, monitoring, and the operational tools to automate deployment, scaling, and lifecycle management of your cluster.
```
Comment (mitechie) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: missing comma after "etcd nodes"

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+- Open the K8s dashboard.
+- Control your cluster from the kubectl CLI client.
+- Deploy your first container workload.
+- Add extra features to your Kubernetes cluster.
+ 
+ 
+### You will need...
+ 
+* An [Ubuntu One account](https://login.ubuntu.com/).
+* An SSH key in your JAAS account. If you haven’t done this already, [learn how to add your SSH key](https://jujucharms.com/docs/2.1/users-auth#credentials-and-ssh-keys).
```
Comment (mitechie) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: "An SSH key"

It's part of the deployment flow so I'd leave off the JAAS bit as you're submitting it during the deployment flow.

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+* Basic knowledge of command line.
+ 
+ 
+ 
+## Deploying your model with JAAS
+Duration: 1:00
+ 
+With JAAS, you don’t need to deal with the setup and configuration of Juju. All you will need is your preferred cloud credentials.
```
Comment (mitechie) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: "All you will need is cloud credentials to AWS, GCE, or Azure."

Comment (mitechie) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: Preferred cloud credentials isn't specific enough. Our audience is technical and knows what AWS/GCE/Azure are without going through the long version.

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+ 
+You will need to deploy [The Canonical Distribution of Kubernetes](https://jujucharms.com/new?dd=bundle/canonical-kubernetes) before you get started.
```
Comment (mitechie) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: The tutorial should walk through the deployment. We discussed having headings that we can link to in the tutorial for things like our emails.

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+ 
+ 
+## Getting started
+Duration: 5:00
+ 
+### Install the Juju client
+ 
+**You will need to have Juju installed locally to operate your cluster. Skip this step if you already have it.**
+ 
+Juju is available as a client on many platforms and distributions. Regardless of your operating system, getting all the required tools is very easy. Visit the [install docs](https://jujucharms.com/docs/stable/reference-install#getting-the-latest-juju) to get the latest version of Juju on **macOS**, **Windows** or **CentOS**.
```
Comment (mitechie) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: Take out the "Regardless of your OS, getting all the required tools is very easy"

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+ 
+2. Install Juju to get the command line client. 
+`$ sudo snap install juju --classic`
+ 
+3. Verify you can run Juju. You will see a summary and a list of common commands.
+`$ juju`
+ 
+You’re done! To learn more on how to get started with Juju, visit the [documentation](https://jujucharms.com/docs/stable/getting-started). 
+ 
+For general help in the CLI:
```
Comment (mitechie) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: Let's collapse this with the `$ juju` test run command and get a two for one.

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+ 
+## Operating your Kubernetes cluster 
+Duration: 5:00
+ 
+1. Fetch the credentials from the Kubernetes Master charm. 
+`$ mkdir -p ~/.kube`
+
+2. Then run:
+`$ juju scp kubernetes-master/0:config ~/.kube/config`
+This might open a separate modal window, in which you should enter the passphrase for your SSH key.
```
Comment (mitechie) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: see, this is the thing. We've skipped installing so now this has a bunch of hard to debug assumptions in it. "passphrase for your SSH key"? what key?

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+![Passphrase window](./images/passphrase-window.png)
+ 
+3. Establish a secure proxy to the apiserver with the command:
```
Comment (mitechie) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: " to the kubernetes apiserver"

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+`$ kubectl proxy`
+ 
+4. Navigate to the Kubernetes dashboard UI through the proxy **xxx.xxx.xxx.xxx:8001/ui**.
```
Comment (mitechie) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: Need a method to check what IP address to use/get.

**Comment by mitechie** [comment]
I've tried to push up a branch to this but I don't have access rights. I then forked it and tried and GH gave me permission issues still so I put the edited file in:

https://gist.github.com/mitechie/f8d36e096798a8cccf36aa76c90ac2ee

I've added the install step of clicking the dd link, I've added the section on adding the ssh key to getting the kubectl client setup as that's where it's relevent. 

We need to get a high priority bug in to add the ssh-key to the dd flow. The fact that we can't operate our biggest bundles and we can't note about adding the key in the DF seems like a big step back imo.

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md (hunk 2 above)
Comment (anthonydillon) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: Remove comma after `monitoring`.

Comment (anthonydillon) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: If you place scaling after monitoring you won't need a comma before the `and`.

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+---
+ 
+ 
+# Get started with Canonical Kubernetes
+ 
+ 
+## Overview
+Duration: 1:00
+ 
+### About the Canonical Distribution of Kubernetes
```
Comment (anthonydillon) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: The title has `Canonical Kubernetes` and here you refer to `Canonical Distribution of Kubernetes`. Which is it? We should be consistent.

**Comment by pmahnke** [comment]
I have added Rick's copy... I cleaned up a few typos too.

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 12]
```diff
@@ -1,187 +1,179 @@
...
 Duration: 1:00
- 
-With JAAS, you don’t need to deal with the setup and configuration of Juju. All you will need is your preferred cloud credentials.
- 
-You will need to deploy [The Canonical Distribution of Kubernetes](https://jujucharms.com/new?dd=bundle/canonical-kubernetes) before you get started. 
- 
+
+With JAAS, you don’t need to deal with the setup and configuration of Juju. All you will need is credentials for AWS, GCE, or Azure.
+
+Kick off deployment of your Kubernetes cluster by clicking the follow link: [The Canonical Distribution of Kubernetes](https://jujucharms.com/new?dd=bundle/canonical-kubernetes).
```
Comment (elvirafigueras) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: To kick off, [deploy The Canonical Distribution of Kubernetes] (https://jujucharms.com/new?dd=bundle/canonical-kubernetes) (if you haven't done it already)

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md (hunk 6 above)
Comment (cassiocassio) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: To get the latest version of Juju on **macOS**, **Windows** or **CentOS**  see the  [install docs](https://jujucharms.com/docs/stable/reference-install#getting-the-latest-juju).

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 13]
```diff
@@ -1,187 +1,179 @@
...
+2. This action creates a deployment titled *microbots* comprised of 5 replicas defined during the run of the action. It also creates a service named *microbots* which binds an *endpoint*, using all 5 of the microbots pods. Finally, it will create an ingress resource, which points at a **xip.io** domain to simulate a proper DNS service
 3. To see the result of this action, run:
  `$ juju show-action-output [id above]`
 You should get a similar response to:
 ![Address microbot](./images/access-microbot.png)
-**Note**: Your FQDN will be different and contains the address of the cloud instance.
- 
-4. Copy and paste the microbot address into your web browser: **microbot.xxx.xxx.xxx.xip.io**. It’s normal to see a 502/ 503 error during initial application turnup.
+
+Positive
```
Comment (elvirafigueras) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: Remove

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 14]
```diff
@@ -1,187 +1,179 @@
...
- 
+
 Juju is available as a client on many platforms and distributions. Regardless of your operating system, getting all the required tools is very easy. Visit the [install docs](https://jujucharms.com/docs/stable/reference-install#getting-the-latest-juju) to get the latest version of Juju on **macOS**, **Windows** or **CentOS**.
- 
+
 If you are running **Ubuntu**, you can install Juju through the following steps:
- 
-1. It’s helpful to Install Snappy if you don’t have it already. 
+
+1. It’s helpful to Install Snappy if you don’t have it already.
```
Comment (cassiocassio) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: First, install snappy, if you don't have it already:

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 15]
```diff
@@ -1,187 +1,179 @@
...
 `$ sudo snap install juju --classic`
- 
 3. Verify you can run Juju. You will see a summary and a list of common commands.
 `$ juju`
- 
-You’re done! To learn more on how to get started with Juju, visit the [documentation](https://jujucharms.com/docs/stable/getting-started). 
- 
-For general help in the CLI: 
+
+You’re done! To learn more on how to get started with Juju, visit the [documentation](https://jujucharms.com/docs/stable/getting-started).
```
Comment (cassiocassio) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: remove :"You're done" - they've only just started...

Learn more about [getting started with Juju](....)

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 16]
```diff
@@ -1,187 +1,179 @@
...
+
+For general help in the CLI:
```
Comment (cassiocassio) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: delete "general"

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 17]
```diff
@@ -1,187 +1,179 @@
...
+needed to speak to your cluster. The first step to getting the config is to
+make sure you have an SSH key in your JAAS deployment of your cluster.
+
+1. To connect to JAAS from the command line you'll need to register with the JAAS controller. You’ll only need to do this the first time.
+`$ juju register jimm.jujucharms.com`
+2. This command will open a new window in your browser. Use Ubuntu SSO to login and authorise your account
+3. You will then be asked to enter a descriptive name for the JAAS controller.  We suggest using *jaas*
+4. You can now import you SSH key from Launchpad or Github into your cluster. For instance, a user with the Github username 'ghuser' would use the following command:
+`$ juju import-ssh-key gh:ghuser`
+5. Verify your SSH key is working by attempting to run the date command on the
```
Comment (elvirafigueras) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: Verify your SSH key is working by running the date command on the first machine

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 18]
```diff
@@ -1,187 +1,179 @@
...
-2. Then run:
+To operate your cluster with kubectl you need to download the configuration
+needed to speak to your cluster. The first step to getting the config is to
+make sure you have an SSH key in your JAAS deployment of your cluster.
+
+1. To connect to JAAS from the command line you'll need to register with the JAAS controller. You’ll only need to do this the first time.
+`$ juju register jimm.jujucharms.com`
+2. This command will open a new window in your browser. Use Ubuntu SSO to login and authorise your account
+3. You will then be asked to enter a descriptive name for the JAAS controller.  We suggest using *jaas*
+4. You can now import you SSH key from Launchpad or Github into your cluster. For instance, a user with the Github username 'ghuser' would use the following command:
```
Comment (cassiocassio) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: add "– if you didn't add it earlier." after "your cluster".

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 19]
```diff
@@ -1,187 +1,179 @@
...
+`$ juju import-ssh-key gh:ghuser`
+5. Verify your SSH key is working by attempting to run the date command on the
+   first machine
+`$ juju run --machine 0 -- date
```
Comment (elvirafigueras) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: `$ juju run --machine 0 -- date` (add tick)

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 20]
```diff
@@ -1,187 +1,179 @@
...
+With SSH operating properly you can now setup *kubectl*. Learn more by visiting the [official documentation](https://kubernetes.io/docs/user-guide/kubectl/)
+6. Fetch the credentials from the Kubernetes Master charm
+`$ mkdir -p ~/.kube`
```
Comment (cassiocassio) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: Move this into step 7 as "Make a directory: `$ mkdir -p ~/.kube` and then copy the config file `$ juju scp kubernetes-master/0:config ~/.kube/config`

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 21]
```diff
@@ -1,187 +1,179 @@
...
+This might open a separate modal window, in which you should enter the passphrase for your SSH key you added in step 4
 ![Passphrase window](./images/passphrase-window.png)
- 
-3. Establish a secure proxy to the apiserver with the command:
+8. Establish a secure proxy to the apiserver with the command:
 `$ kubectl proxy`
- 
-4. Navigate to the Kubernetes dashboard UI through the proxy **xxx.xxx.xxx.xxx:8001/ui**.
+
+9. Navigate to the Kubernetes dashboard UI through the proxy **xxx.xxx.xxx.xxx:8001/ui**
```
Comment (cassiocassio) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: Add a sentence to point 9.  "Note the trailing `/ui`"

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 22]
```diff
@@ -0,0 +1,179 @@
...
+categories: jaas
+tags: jaas, juju, kubernetes, beginner, container, docker, cloud
+difficulty: 3
+status: Published
+Published: 2017-06-01
+
+---
+
+
+# Get started with Canonical Kubernetes
```
Comment (anthonydillon) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: s/Canonical Kubernetes/Canonical Distribution of Kubernetes

File: tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md [hunk 23]
```diff
@@ -0,0 +1,179 @@
...
+* Basic knowledge of command line
+
+
+
+## Deploying your model with JAAS
+Duration: 1:00
+
+With JAAS, you don’t need to deal with the setup and configuration of Juju. All you will need is credentials for AWS, GCE, or Azure.
+
+Kick off, [deploy The Canonical Distribution of Kubernetes](https://jujucharms.com/new?dd=bundle/canonical-kubernetes) (if you haven't done this already).
```
Comment (anthonydillon) [inline tutorials/get-started-canonical-kubernetes/get-started-canonical-kubernetes.md]: The dd link creates a lot of errors.

**Pull request merged**